tests/entries/test_compatibility.py

```diff
@@ -80,9 +80,7 @@ def setUp(self):
         self.compat = MaterialsProjectCompatibility(check_potcar_hash=False)
 
     def test_correction_specificity(self):
-        processed = self.compat.process_entries([self.entry1, self.entry2, self.entry3])
-
-        assert len(processed) == 2
```

Comment:
Please help me check carefully here @rkingsbury as I'm not 100% confident about these two changes in unit tests. Prior to this change: https://github.com/materialsproject/pymatgen/blob/8aa1cf89a93e12c2ac16ddde517105e307a04b91/pymatgen/entries/compatibility.py#L468-L482
if run_type is not in {"GGA", "GGA+U"}, an error would be raise thus no correction would be applied. After this change, a correction of 0.0 would be returned thus the number of entries would increase from 2 to 3. Correct me if I'm wrong. Thanks.

Reply:
Great catch @DanielYang59 . We should not make a change that causes these tests to fail (for backward compatibility reasons).

I've studied the code again, and actually you are right that the `raise` statement should remain.

The logic is the following:
- when `process_entries` is called, it calls `get_adjustments` for every entry provided
- Any entry which raises a `CompatibilityError` is excluded from the returned list.
- In the specific case of `MaterialsProjectCompatibility`, `get_adjustments` calls `get_corrections_dict`, which loop through the `get_correction` method of all 4 of the underlying `Correction` classes (we eliminated these in the newer scheme for simplicity)
- It just so happens that the `raise` statement that occurs in `UCorrection` (which processes any entry sent through `get_Adjustments`) serves the purpose of excluding incompatible entries. 
- Strictly speaking, this `raise` statement could/should occur in `get_adjustments` itself, because that is really the only place that the run_type check needs to happen (an entry which is incompatible with one of the corrections is incompatible with all of them, for this purpose)
- The `return ufloat(0,0)` statements in the `Correction` classes do not serve an important purpose as far as I can tell.

I think the cleanest fix would be to move the run type check and `raise` statement to `get_adjustments` itself and remove the other statements, as illustrated [here](https://github.com/materialsproject/pymatgen/pull/3540/files)

Reply:
Tests pass for me after making that change